internal/registry/registry.go

```diff
@@ -391,7 +391,12 @@ func (r *Registry) RegisterOrUpdate(ctx context.Context, providerID peer.ID, add
 	}
 
 	if info.Publisher.Validate() == nil && info.PublisherAddr == nil && info.Publisher == info.AddrInfo.ID {
-		info.PublisherAddr = info.AddrInfo.Addrs[0]
+		if len(info.AddrInfo.Addrs) == 0 {
+			log.Warnw("Register provider with no provider or publisher addresses",
+				"provider", info.AddrInfo.ID, "publisher", info.Publisher)
+		} else {
+			info.PublisherAddr = info.AddrInfo.Addrs[0]
```

Comment:
file a low importance bug to be able to support multiple addresses round-robin :)

Reply:
@willscott #321